File: backend/app/models/query.py
from typing import Any

from pydantic import Field

from app.models import AppBaseModel


class QueryRequest(AppBaseModel):
    sql: str


class QueryColumn(AppBaseModel):
    name: str
    data_type: str


class QueryResult(AppBaseModel):
    sql: str
    columns: list[QueryColumn]
    rows: list[dict[str, Any]]
    row_count: int
    truncated: bool
    elapsed_ms: int


class NaturalQueryRequest(AppBaseModel):
    prompt: str = Field(..., min_length=1)


class NaturalQueryResult(AppBaseModel):
    generated_sql: str


class ErrorResponse(AppBaseModel):
    error: str
    message: str
    detail: str | None = None

File: frontend/src/types/query.ts
/** 查询 API 类型，与 contracts/api.md 一致 */

export interface QueryRequest {
  sql: string;
}

export interface QueryColumn {
  name: string;
  dataType: string;
}

export interface QueryResult {
  sql: string;
  columns: QueryColumn[];
  rows: Record<string, unknown>[];
  rowCount: number;
  truncated: boolean;
  elapsedMs: number;
}

/** 后端错误体（扁平 JSON，非 FastAPI 默认 detail 包裹） */
export interface ApiErrorBody {
  error: string;
  message: string;
  detail?: string | null;
}
